app/test_main.py

```diff
@@ -1 +1,53 @@
-# write your code here
+from unittest import mock
+from app.main import can_access_google_page
+from typing import Callable
+import pytest
+
+
+@pytest.fixture()
+def mocked_url_validator() -> None:
+    with mock.patch("app.main.valid_google_url") as mocked_validator:
+        yield mocked_validator
+
+
+@pytest.fixture()
+def mocked_internet_connection() -> None:
+    with mock.patch("app.main.has_internet_connection") as mocked_connection:
+        yield mocked_connection
+
+
+def test_valid_url_and_connection(
+        mocked_url_validator: Callable,
+        mocked_internet_connection: Callable
+) -> None:
+    mocked_internet_connection.return_value = True
```

Comment:
You have almost identical tests, only arguments are different. Use @pytest.mark.parametrize